sympy/solvers/solveset.py

```diff
@@ -2079,16 +2081,25 @@ def solveset(f, symbol=None, domain=S.Complexes):
         # the xreplace will be needed if a ConditionSet is returned
         return solveset(f[0], s[0], domain).xreplace(swap)
 
-    if domain.is_subset(S.Reals):
-        if not symbol.is_real:
-            assumptions = symbol.assumptions0
-            assumptions['real'] = True
-            try:
-                r = Dummy('r', **assumptions)
-                return solveset(f.xreplace({symbol: r}), r, domain
-                    ).xreplace({r: symbol})
-            except InconsistentAssumptions:
-                pass
+    # solveset should ignore assumptions on symbols
+    s = symbol.as_set()
+    rc = [{'real': True}, {'complex': True}]
```

Comment:
It seems that this checks the assumptions on the symbol and replaces the symbol if the assumptions are inconsistent with the domain. I think it would be better to guarantee always replacing the symbol with either a symbol that has no assumptions or with one whose assumptions are derived solely from the domain so that we can confidently say that the assumptions on the unknown symbols are being ignored.